### PR TITLE
Document EKSClusterConfig API

### PR DESCRIPTION
**Pull request by mjura**
Issue: https://github.com/rancher/eks-operator/issues/203

### CODE REVIEW

File: pkg/apis/eks.cattle.io/v1/types.go [hunk 1]
```diff
@@ -33,63 +33,111 @@ type EKSClusterConfig struct {
...
+	// AmazonCredentialSecret is the name of the secret containing the Amazon credentials.
+	AmazonCredentialSecret string `json:"amazonCredentialSecret"`
+	// DisplayName is the name of the cluster to be displayed in the UI.
+	DisplayName string `json:"displayName" norman:"noupdate"`
+	// Region is the AWS region to create the cluster in.
+	Region string `json:"region" norman:"noupdate"`
+	// Imported is true if the cluster was imported.
+	Imported bool `json:"imported" norman:"noupdate"`
+	// KubernetesVersion is the version of Kubernetes to use.
+	KubernetesVersion *string `json:"kubernetesVersion" norman:"pointer"`
```
Comment (richardcase) [inline pkg/apis/eks.cattle.io/v1/types.go:45]: Normally with kubebuilder/controller runtime a pointer indicates that a field is optional. If thats the case for the pointer values here we can `+optional`

File: pkg/apis/eks.cattle.io/v1/types.go [hunk 2]
```diff
@@ -33,63 +33,111 @@ type EKSClusterConfig struct {
...
+	NodeGroups []NodeGroup `json:"nodeGroups"`
 }
 
 type EKSClusterConfigStatus struct {
-	Phase                         string            `json:"phase"`
-	VirtualNetwork                string            `json:"virtualNetwork"`
-	Subnets                       []string          `json:"subnets"`
-	SecurityGroups                []string          `json:"securityGroups"`
-	ManagedLaunchTemplateID       string            `json:"managedLaunchTemplateID"`
+	// Phase is the current lifecycle phase of the cluster.
```
Comment (richardcase) [inline pkg/apis/eks.cattle.io/v1/types.go:73]: It would also be good to add doc strings to the struct definitions.

**Review by richardcase — COMMENTED** [review]
Thanks @mjura 👍  Just a few minor comments.

File: pkg/apis/eks.cattle.io/v1/types.go (hunk 2 above)
Comment (mjura) [reply pkg/apis/eks.cattle.io/v1/types.go:95]: fixed

File: pkg/apis/eks.cattle.io/v1/types.go [hunk 3]
```diff
@@ -33,63 +37,156 @@ type EKSClusterConfig struct {
...
+	// +kubebuilder:validation:UniqueItems:=true
+	Tags map[string]string `json:"tags"`
+	// SecretsEncryption is true if secrets should be encrypted.
+	// +optional
+	SecretsEncryption *bool `json:"secretsEncryption" norman:"noupdate"`
+	// KmsKey is the KMS key to use for encryption.
+	// +optional
+	KmsKey *string `json:"kmsKey" norman:"noupdate,pointer"`
+	// PublicAccess is true if the cluster should be publicly accessible.
+	// +kubebuilder:validation:Required
```
Comment (richardcase) [inline pkg/apis/eks.cattle.io/v1/types.go:66]: Are these required, even though they are pointer values?

**Review by richardcase — COMMENTED** [review]
@mjura - are you sure that everything marked as "required" is actually required? The reason i ask is that a lof of the types are pointers which can denote something being optional.

File: pkg/apis/eks.cattle.io/v1/types.go (hunk 3 above)
Comment (mjura) [reply pkg/apis/eks.cattle.io/v1/types.go:66]: fixed

**Comment by mjura** [comment]
> @mjura - are you sure that everything marked as "required" is actually required? The reason i ask is that a lof of the types are pointers which can denote something being optional.

I think that pointers should be verified, but let's stay consistent with Kubebuilder markers

**Review by salasberryfin — APPROVED** [review]

**Pull request merged**